snippets.py

```diff
@@ -371,311 +380,318 @@ def post(self):
-                                              1000 if y[3] is None else y[3])
-                                         or cmp(x[0], y[0])))
+@app.route("/admin/manage_users", methods=["GET", "POST"])
+def admin_manage_users_handler():
+    """Lets admins delete and otherwise manage users."""
+    # options are 'email', 'creation_time', 'last_snippet_time'
+    sort_by = flask.request.args.get('sort_by', 'creation_time')
+
+    # First, check if the user had clicked on a button.
+    if flask.request.form.get("action") == "hide":
+        email_of_user_to_hide = flask.request.form.get("email")
+        # TODO(csilvers): move this get/update/put atomic into a txn
+        user = util.get_user_or_die(email_of_user_to_hide)
+        user.is_hidden = True
+        user.put()
+        logging.info("Hide user: %s", user.email)
+        return flask.redirect('/admin/manage_users?sort_by=%s&msg=%s+hidden'
+                              % (sort_by, user.email))
+    elif flask.request.form.get("action") == "unhide":
+        email_of_user_to_unhide = flask.request.form.get("email")
+        # TODO(csilvers): move this get/update/put atomic into a txn
+        user = util.get_user_or_die(email_of_user_to_unhide)
+        user.is_hidden = False
+        user.put()
+        logging.info("Unhide user: %s", user.email)
+        return flask.redirect('/admin/manage_users?sort_by=%s&msg=%s+unhidden'
+                              % (sort_by, user.email))
+    elif flask.request.form.get("action") == "delete":
+        email_of_user_to_delete = flask.request.form.get("email")
+        user = util.get_user_or_die(email_of_user_to_delete)
+        user.key.delete()
+        logging.info("Delete user: %s", user.email)
+        return flask.redirect('/admin/manage_users?sort_by=%s&msg=%s+deleted'
+                              % (sort_by, user.email))
+
+    user_q = models.User.query()
+    results = user_q.fetch(1000)
+
+    # Tuple: (email, is-hidden, creation-time, days since last snippet)
+    user_data = []
+    for user in results:
+        # Get the last snippet for that user.
+        last_snippet = util.most_recent_snippet_for_user(user.email)
+        if last_snippet:
+            seconds_since_snippet = (
+                (_TODAY_FN().date() - last_snippet.week).total_seconds())
+            weeks_since_snippet = int(
+                seconds_since_snippet /
+                datetime.timedelta(days=7).total_seconds())
         else:
-            raise ValueError('Invalid sort_by value "%s"' % sort_by)
+            weeks_since_snippet = None
+        user_data.append((user.email, user.is_hidden,
+                          user.created, weeks_since_snippet))
+
+    # We have to use 'cmp' here since we want ascending in the
+    # primary key and descending in the secondary key, sometimes.
+    if sort_by == 'email':
+        user_data.sort(key=cmp_to_key(lambda x, y: cmp(x[0], y[0])))
```

Comment:
Do you need all this cmp_to_key/cmp stuff?  Couldn't you just do `key=lambda x: x[0]`?

Reply:
You're right, it looks like that accomplishes the same thing.  Fixing!